### PR TITLE
Increase scan interval to stabilize TestFilestreamCloseRenamed

### LINKED ISSUE #24623: Flaky Test [Build&Test / filebeat-build / TestFilestreamCloseRenamed – github.com/elastic/beats/v7/filebeat/input/filestream]
## Flaky Test

* **Test Name:** `Build&Test / filebeat-build / TestFilestreamCloseRenamed – github.com/elastic/beats/v7/filebeat/input/filestream`
* **Artifact Link:** https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fbeats%2FPR-24560/detail/PR-24560/7/
* **PR:** #24560
* **Commit:** 69822a456deacfce120bc344242240ef32d63a25

### Error details
```
Failed
```

### Stack Trace
```
=== RUN   TestFilestreamCloseRenamed
coverage: 58.2% of statements
panic: test timed out after 10m0s

goroutine 83 [running]:
testing.(*M).startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1618 +0xe5
created by time.goFunc
	/usr/local/go/src/time/sleep.go:167 +0x45

goroutine 1 [chan receive, 9 minutes]:
testing.(*T).Run(0xc000502c00, 0xb19f0d, 0x1a, 0xb369a8, 0x494601)
	/usr/local/go/src/testing/testing.go:1169 +0x2da
testing.runTests.func1(0xc00021cd80)
	/usr/local/go/src/testing/testing.go:1439 +0x78
testing.tRunner(0xc00021cd80, 0xc0002dfda8)
	/usr/local/go/src/testing/testing.go:1123 +0xef
testing.runTests(0xc000235320, 0x10086c0, 0x13, 0x13, 0xc00ce177872e9095, 0x8bb47f838c, 0x1026f60, 0x40f668)
	/usr/local/go/src/testing/testing.go:1437 +0x2fe
testing.(*M).Run(0xc0004a6380, 0x0)
	/usr/local/go/src/testing/testing.go:1345 +0x1eb
main.main()
	_testmain.go:139 +0x1c5

goroutine 67 [sleep]:
time.Sleep(0x989680)
	/usr/local/go/src/runtime/time.go:188 +0xbf
github.com/elastic/beats/v7/filebeat/input/filestream.(*inputTestingEnvironment).waitUntilEventCount(0xc0004a4000, 0x3)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/environment_test.go:237 +0x88
github.com/elastic/beats/v7/filebeat/input/filestream.TestFilestreamCloseRenamed(0xc000502c00)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/input_integration_test.go:69 +0x539
testing.tRunner(0xc000502c00, 0xb369a8)
	/usr/local/go/src/testing/testing.go:1123 +0xef
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:1168 +0x2b3

goroutine 72 [semacquire, 9 minutes]:
sync.runtime_Semacquire(0xc00009c538)
	/usr/local/go/src/runtime/sema.go:56 +0x45
sync.(*WaitGroup).Wait(0xc00009c530)
	/usr/local/go/src/sync/waitgroup.go:130 +0x65
github.com/elastic/go-concert/unison.(*MultiErrGroup).Wait(0xc00009c510, 0x0, 0x0, 0x0)
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:54 +0x53
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileProspector).Run(0xc0002a7e00, 0xc000098fe0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/prospector.go:190 +0x3a5
github.com/elastic/beats/v7/filebeat/input/filestream/internal/input-logfile.(*managedInput).Run(0xc0002a7e50, 0xc000098fe0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/internal/input-logfile/input.go:76 +0x36b
github.com/elastic/beats/v7/filebeat/input/filestream.(*inputTestingEnvironment).startInput.func1(0xbcd8a0, 0xc0002293c0, 0xbcb5a0, 0xc0002a7e50, 0xc0004a4000, 0xc0004a4090)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/environment_test.go:92 +0x14c
created by github.com/elastic/beats/v7/filebeat/input/filestream.(*inputTestingEnvironment).startInput
	/go/src/github.com/elastic/beats/filebeat/input/filestream/environment_test.go:88 +0x9a

goroutine 9 [select]:
github.com/elastic/go-concert/timed.Periodic(0x7fb474082030, 0xc0000aa040, 0xf4240, 0xc0000ddec0, 0x0, 0x0)
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/timed/timed.go:76 +0x13d
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileWatcher).Run(0xc000155da0, 0x7fb474082008, 0xc0000aa040)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/fswatch.go:122 +0x10b
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileProspector).Run.func1(0x0, 0x0)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/prospector.go:99 +0x9a
github.com/elastic/go-concert/unison.(*MultiErrGroup).Go.func1(0xc00009c510, 0xc00040a000)
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:42 +0x64
created by github.com/elastic/go-concert/unison.(*MultiErrGroup).Go
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:40 +0x66

goroutine 10 [chan receive, 9 minutes]:
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileWatcher).Event(0xc000155da0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/fswatch.go:210 +0x85
github.com/elastic/beats/v7/filebeat/input/filestream.(*fileProspector).Run.func2(0x0, 0x0)
	/go/src/github.com/elastic/beats/filebeat/input/filestream/prospector.go:105 +0xef
github.com/elastic/go-concert/unison.(*MultiErrGroup).Go.func1(0xc00009c510, 0xc0000aa080)
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:42 +0x64
created by github.com/elastic/go-concert/unison.(*MultiErrGroup).Go
	/go/pkg/mod/github.com/elastic/go-concert@v0.1.0/unison/multierrgroup.go:40 +0x66

```

### PR DESCRIPTION
**Pull request by kvch**
Closes #24623

### CODE REVIEW

File: filebeat/input/filestream/input_integration_test.go [hunk 1]
```diff
@@ -45,17 +45,17 @@ func TestFilestreamCloseRenamed(t *testing.T) {
...
 		"close.on_state_change.renamed":        "true",
 	})
 
-	ctx, cancelInput := context.WithCancel(context.Background())
-	env.startInput(ctx, inp)
-
 	testlines := []byte("first log line\n")
 	env.mustWriteLinesToFile(testlogName, testlines)
 
+	ctx, cancelInput := context.WithCancel(context.Background())
```
Comment (ruflin) [inline filebeat/input/filestream/input_integration_test.go:56]: For my understanding, if the input is closed it triggers sending the pending events?

File: filebeat/input/filestream/input_integration_test.go [hunk 2]
```diff
@@ -45,17 +45,17 @@ func TestFilestreamCloseRenamed(t *testing.T) {
 	testlogName := "test.log"
 	inp := env.mustCreateInput(map[string]interface{}{
 		"paths":                                []string{env.abspath(testlogName) + "*"},
-		"prospector.scanner.check_interval":    "1ms",
+		"prospector.scanner.check_interval":    "10ms",
```
Comment (ruflin) [inline filebeat/input/filestream/input_integration_test.go:48]: How does this change fix the test?

**Review by ruflin — COMMENTED** [review]
I would like to understand a bit better how the change fixes the test. Sorry, new to the code base.

File: filebeat/input/filestream/input_integration_test.go (hunk 1 above)
Comment (kvch) [reply filebeat/input/filestream/input_integration_test.go:56]: No. If the input is closed the whole reading and publishing pipeline is cancelled. That is why first the test waits for all expected events to arrive and then the input is cancelled.

File: filebeat/input/filestream/input_integration_test.go (hunk 2 above)
Comment (kvch) [reply filebeat/input/filestream/input_integration_test.go:48]: The problem was that sometimes the prospector detected the rename earlier than the harvester and cancelled the reading before the event could have been sent. By increasing the interval, it is guaranteed that the harvester detects the rename.

Comment (ruflin) [reply filebeat/input/filestream/input_integration_test.go:48]: 👍

**Review by ruflin — APPROVED** [review]
Thanks for the details @kvch I must confess I read at first `10min` instead of `10ms` 🤦‍♂️

File: filebeat/input/filestream/input_integration_test.go (hunk 2 above)
Comment (michalpristas) [reply filebeat/input/filestream/input_integration_test.go:48]: can you put this into a description?

**Review by michalpristas — APPROVED** [review]
please fill out description with what you have in comments so it is clear how this helps

**Comment by kvch** [comment]
jenkins run tests

**Comment by kvch** [comment]
jenkins run tests

**Comment by kvch** [comment]
jenkins run tests

**Pull request merged**